waku/protocol/v2/waku_store.nim

```diff
@@ -0,0 +1,130 @@
+import chronos, chronicles
+import ./filter
+import tables
+import libp2p/protocols/pubsub/pubsub,
+       libp2p/protocols/pubsub/pubsubpeer,
+       libp2p/protocols/pubsub/floodsub,
+       libp2p/protocols/pubsub/gossipsub,
+       libp2p/protocols/pubsub/rpc/[messages, protobuf],
+       libp2p/protocols/protocol,
+       libp2p/protobuf/minprotobuf,
+       libp2p/stream/connection
+
+import metrics
+
+import stew/results
+
+const
+  WakuStoreCodec* = "/vac/waku/store/2.0.0-alpha2"
+
+type
+  StoreRPC* = object
+    query*: seq[HistoryQuery]
+    response*: seq[HistoryResponse] 
+
+  HistoryQuery* = object
```

Comment:
I'd put exactly these in the spec https://github.com/vacp2p/specs/issues/166

Is there a follow up issue of todos when it comes to enhancements like windows, etc? What is the minimum we need to support initially?